File: enternal-rune-user-frontend/src/types/ProductVariant.ts
import { Image } from "./Image";
import { ProductPrice } from "./ProductPrice";

export type ProductVariant = {
    prodvId: number;
    prodvProduct: number;
    prodvName: string;
    prodvImage: Image;
    prodvModel: string;
    prodvVersion: string;
    prodvColor: string;
    prodvPrice: ProductPrice;
};

File: enternal-rune-backend/src/main/java/iuh/fit/se/enternalrunebackend/entity/ProductVariant.java
package iuh.fit.se.enternalrunebackend.entity;

import jakarta.persistence.*;
import lombok.*;
import lombok.experimental.FieldDefaults;

@Entity
@NoArgsConstructor
@AllArgsConstructor
@Getter
@Setter
@Table(name = "product_variants")
@FieldDefaults(level = AccessLevel.PRIVATE)
public class ProductVariant {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "product_variant_id")
    Long prodvId;

    @ManyToOne
    Product prodvProduct;

    String prodvName;

    String prodvModel;

    String prodvVersion;

    String prodvColor;

    @ManyToOne
    @JoinColumn(name = "pp_price")
    ProductPrice prodvPrice;

    @ManyToOne
    @JoinColumn(name = "image_id")
    Image prodvImg;
}
